packages/destination-actions/src/destinations/salesforce/sf-operations.ts

```diff
@@ -74,22 +67,30 @@ export default class Salesforce {
   }
 
   private baseUpdate = async (recordId: string, sobject: string, payload: GenericPayload) => {
+    const JSON_data = this.buildJSONData(payload, sobject)
+
     return this.request(`${this.instanceUrl}/services/data/${API_VERSION}/sobjects/${sobject}/${recordId}`, {
       method: 'patch',
-      json: {
-        LastName: payload.last_name,
-        Company: payload.company,
-        FirstName: payload.first_name,
-        State: payload.state,
-        Street: payload.street,
-        Country: payload.country,
-        PostalCode: payload.postal_code,
-        City: payload.city,
-        Email: payload.email
-      }
+      json: JSON_data
     })
   }
 
+  private buildJSONData = (payload: GenericPayload, sobject: string) => {
+    let baseShape
+    if (payload.sobject) {
+      //sobject is a unique key for the custom object action
+      baseShape = {}
+    } else {
+      baseShape = mapObjectToShape(payload, sobject)
+    }
+
+    if (payload.custom_fields) {
+      baseShape = { ...baseShape, ...payload.custom_fields }
```

Comment:
```suggestion
      // custom field mappings take priority over base shape mappings.
      baseShape = { ...baseShape, ...payload.custom_fields }
```

Just a  note, this ordering implies that any `custom_fields` mappings will take priority over any `baseShape` mappings. That may  be  expected but wanted to  call that out. Might be  worth a comment.